src/Provider.tsx

```diff
@@ -17,7 +17,7 @@ export declare interface ProviderProps {
   client: ApolloClient;
 }
 
-export default class Provider extends Component<ProviderProps, any> {
+export default class ApolloProvider extends Component<ProviderProps, any> {
```

Comment:
Yeah those are the kinda questions I wanted to talk through. Since we added some duplication with this function. 

I think `ApolloProvider` should be shipped using `composeApolloProvider` under the hood.